core/src/main/java/org/apache/iceberg/SchemaUpdate.java

```diff
@@ -192,6 +193,92 @@ public UpdateSchema updateColumnDoc(String name, String doc) {
     return this;
   }
 
+  @Override
+  public UpdateSchema updateSchema(Schema newSchema) {
+    TypeUtil.visit(newSchema, new ApplyUpdates(schema, this));
```

Comment:
Should we reassign field ids here based on the old schema? ApplyUpdates requires field ids to match the old schema for correct behaviour.

Reply:
Any precondition checks regarding the unsupported operations can also go here.

Reply:
I think, by the time schema reaches this point it should already be assigned correct ids. Lets see how Spark does it. We can then revisit thi

Reply:
Sure. We can take a look later.